AddQtAndroidApk.cmake

```diff
@@ -97,84 +98,162 @@ macro(add_qt_android_apk TARGET SOURCE_TARGET)
             set(QT_ANDROID_SDK_BUILDTOOLS_REVISION ${BUILD_TOOLS_VERSION})
         endif()
     endforeach()
-    message("Detected Android SDK build tools version ${QT_ANDROID_SDK_BUILDTOOLS_REVISION}")
 
-    # define the application source package directory
-    if(ARG_PACKAGE_SOURCES)
-        set(QT_ANDROID_APP_PACKAGE_SOURCE_ROOT ${ARG_PACKAGE_SOURCES})
+    if(ARG_VERBOSE)
+        message(STATUS "Detected Android SDK build tools version ${QT_ANDROID_SDK_BUILDTOOLS_REVISION}")
+    endif()
+
+    if(ARGS_EXTRA_QML)
+        set(QT_ANDROID_QML_IMPORT_PATH ${ARGS_EXTRA_QML})
+    endif()
+
+    set(QT_ANDROID_APP_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/${SOURCE_TARGET}-${ANDROID_ABI})
+    if(ARG_ANDROID_MANIFEST_IN_PATH)
+        set(QT_ANDROID_MANIFEST_IN_REAL_PATH ${ARG_ANDROID_MANIFEST_IN_PATH})
     else()
-        # get version code from arguments, or generate a fixed one if not provided
-        set(QT_ANDROID_APP_VERSION_CODE ${ARG_VERSION_CODE})
-        if(NOT QT_ANDROID_APP_VERSION_CODE)
-            set(QT_ANDROID_APP_VERSION_CODE 1)
-        endif()
+        set(QT_ANDROID_MANIFEST_IN_REAL_PATH ${QT_ANDROID_SOURCE_DIR}/AndroidManifest.xml.in)
+    endif()
 
+    if(ARG_VERBOSE)
+        message(STATUS "Used input AndroidManifest file QT_ANDROID_MANIFEST_IN_REAL_PATH : ${QT_ANDROID_MANIFEST_IN_REAL_PATH}")
+    endif()
+
+    # get version code from arguments, or generate a fixed one if not provided
+    set(QT_ANDROID_APP_VERSION_CODE ${ARG_VERSION_CODE})
+    if(NOT QT_ANDROID_APP_VERSION_CODE)
+        set(QT_ANDROID_APP_VERSION_CODE 1)
+    endif()
+
+    if(ARG_VERSION_NAME)
+        set(QT_ANDROID_APP_VERSION ${ARG_VERSION_NAME})
+        if(NOT QT_ANDROID_APP_VERSION)
+            set(QT_ANDROID_APP_VERSION 1)
+        endif()
+    else(ARG_VERSION_NAME)
         # try to extract the app version from the target properties, or use the version code if not provided
         get_property(QT_ANDROID_APP_VERSION TARGET ${SOURCE_TARGET} PROPERTY VERSION)
         if(NOT QT_ANDROID_APP_VERSION)
             set(QT_ANDROID_APP_VERSION ${QT_ANDROID_APP_VERSION_CODE})
         endif()
+    endif(ARG_VERSION_NAME)
```

Comment:
About this new `VERSION_NAME` option: why would the Android APK have a different version than the project itself? do you have a relevant use case for this addition?

Reply:
I've added this option because at the time i didn't use the `VERSION` property of project. And i think it's nice to expose, for example if you have your internal version used in your project. And a version numbering different for the customer.
Or if you want to expose only a part of the version to your customer. Anyway this field is mostly decorative i think(i might be wrong). But from what i understand, it's the VERSION_CODE that is important to android.

Reply:
If this new option is not really necessary, I prefer to keep things simple. Adding another `VERSION_XXX` variable on top of `VERSION_CODE` and CMake's `PROJECT_VERSION` might be a little confusing. And to me this variable should not be different from the project's version, that's exactly the same thing.

This project is just maintained for my own needs, so I like to keep it "minimal", so that I'm still able to work on it when I need to, even after 2 or 3 years without touching it 😊

Reply:
Yes i see your point, I should start using PROJECT_VERSION and not doing any workaround.

Reply:
This variable is a string so it can be used to decorate the version in Android. For example doing something like "v1.0 beta" or "v1.0 rc1". I don't think you can do such thing with `PROJECT_VERSION`?

Reply:
I don't know, but please don't spend so much time searching artificial use cases. Let's stick to what we need right now, and don't worry, if someone wants to support "beta" or "rc" in its Android version string, I'm sure he'll find hiw way to this github repo 😉